src/lib/protocols/ajp.c

```diff
@@ -110,12 +110,6 @@ static void ndpi_check_ajp(struct ndpi_detection_module_struct *ndpi_struct,
 void ndpi_search_ajp(struct ndpi_detection_module_struct *ndpi_struct,
  struct ndpi_flow_struct *flow)
 {
-  // Break after 20 packets.
-  if(flow->packet_counter > 20) {
```

Comment:
Are you sure that this is 0xdeadc0de or just not triggered by any pcap file?
I can imagine that the side effects of other dissectors may it look like this code is dead, but is it the case in a "real" environment?

Reply:
All the code paths in `ndpi_check_ajp()` leads to `set_ajp_detected()` or to `NDPI_EXCLUDE_PROTO()`
In other words, `ndpi_search_ajp()` is called only for the first packet of the flow.
So I do think that this "if block" is never triggered